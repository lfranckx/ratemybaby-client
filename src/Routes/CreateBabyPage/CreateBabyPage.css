#create-section {
    display: inline-block;
    background-color: rgb(249, 103, 103);
    background-image: linear-gradient(to bottom right, rgb(247, 87, 87) , rgb(249, 147, 103));    
    border-radius: 10px;
    padding: 40px;
    box-shadow: 1.5px 3px 5px grey;
    width: 400px;
}

#create-form > * {
    margin: 10px;
}

#create-header {
    color: white;
    font-size: 2rem;
    opacity: 0.9;  
}

#create-section label {
    color: white;
}

.form-items {
    text-align: left;
}

.create-form-label {
    font-size: 18px;
    margin-left: 3px;
}

.create-input {
    font-size: 16px;
    border-radius: 8px;
    box-shadow: 1.5px 2px 3px grey;
    padding: 5px 10px;
}

#country-selector {
    font-size: 16px;
    border-radius: 8px;
    padding: 5px;
    width: 300px;
}

.create-about {
    font-size: 16px;
    border: none;
    border-radius: 8px;
    padding: 10px;
    box-shadow: 1.5px 2px 3px grey;
    white-space: pre-wrap;
}

.submit-item {
    text-align: center;
}

#create-button {
    background: white;
    color: rgb(249, 103, 103);
    box-shadow: 1.5px 2px 3px grey;
    display: inline-block;
    margin-top: 10px;
    padding: 10px 30px;
    border-radius: 50px;
    border: none;
    cursor: pointer;
}

#create-button:active {
    transform: translateY(2px);
}

.selector {
    box-shadow: 1.5px 1px 1px grey;
}

/* MEDIA QUERIES */

@media (max-width: 700px) {
    #create-section {
        width: 320px;
    }

    #create-header {
        font-size: 1.5rem;
    }

    #create-form {
        font-size: 1rem;
    }

    #create-form input {
        font-size: 10px;
    }

    #country-selector {
        width: 220px;
    }

    #login-button {
        font-size: 15px;
        padding: 6.5px 35px;
    }

    #create-button {
        padding: 6.5px 35px;
    }
}