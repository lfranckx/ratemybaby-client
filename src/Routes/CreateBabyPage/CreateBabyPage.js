import React, { Component } from 'react'
import CreateProfileForm from '../../Components/CreateProfileForm/CreateProfileForm'
import BabyContext from '../../Contexts/BabyContext'

export default class EditProfile extends Component {
    static defaultProps = {
        location: {},
        history: {
            push:() => {}
        }
    }
    
    static contextType = BabyContext

    handleSubmitForm = () => {
        console.log('successfully creating baby profile');

        const { location, history } = this.props
        const destination = (location.state || {}).from || '/uploadimage'
        history.push(destination)
    }

    render() {
        const { user } = this.context        
        return (
            <section>
                <h3>Create Baby Profile</h3>
                <CreateProfileForm 
                    onSubmitForm={this.handleSubmitForm} 
                    user={user}
                />
            </section>
        )
    }
}