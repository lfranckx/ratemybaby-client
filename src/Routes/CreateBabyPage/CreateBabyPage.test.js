import React from 'react';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import CreateBabyPage from './CreateBabyPage';

const nullBaby = {
    baby: null
}

const context = {
    active: false,
    usersBabies: [],
    babies: [],
    baby: nullBaby,
    error: null,
    setError: () => {},
    clearError: () => {},
    toggleActive: () => {},
    setNotActive: () => {},
    setBaby: () => {},
    clearBaby: () => {},
    setUsersBabies: () => {}
}

describe(`CreateBabyPage component`, () => {
    it(`renders page without crashing`, () => {
        const wrapper = mount(<CreateBabyPage />, { context: context })
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    
})