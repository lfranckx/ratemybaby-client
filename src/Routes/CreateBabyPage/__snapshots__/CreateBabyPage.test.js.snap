// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBabyPage component renders page without crashing 1`] = `
<EditProfile
  history={
    Object {
      "push": [Function],
    }
  }
  location={Object {}}
>
  <section
    id="create-section"
  >
    <h2
      id="create-header"
    >
      Create Baby Profile
    </h2>
    <EditProfile
      onSubmitForm={[Function]}
    >
      <form
        id="create-form"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <div
          className="form-items"
        >
          <label
            className="create-form-label"
          >
            Name
          </label>
        </div>
        <div
          className="form-items"
        >
          <input
            aria-label="name"
            className="create-input"
            id="name"
            name="name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-items"
        >
          <label
            className="create-form-label"
          >
            Age
          </label>
        </div>
        <div
          className="form-items"
        >
          <input
            aria-label="age"
            className="create-input"
            id="age"
            min="1"
            name="age"
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-items"
        >
          <select
            aria-label="format"
            className="month-year selector"
            id="format"
            name="format"
          >
            <option>
              month
            </option>
            <option>
              months
            </option>
            <option>
              year
            </option>
            <option>
              years
            </option>
          </select>
        </div>
        <div
          className="form-items"
        >
          <label
            className="create-form-label"
          >
            Country
          </label>
        </div>
        <div
          className="form-items"
        >
          <input
            aria-label="country"
            className="edit-input"
            id="country"
            name="country"
          />
        </div>
        <div
          className="form-items"
        >
          <label
            className="create-form-label"
          >
            About
          </label>
        </div>
        <div
          className="form-items"
        >
          <textarea
            aria-label="about"
            className="width-100 create-about"
            id="about"
            name="about"
            rows="10"
          />
        </div>
        <div
          className="submit-item"
        >
          <button
            id="create-button"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </EditProfile>
  </section>
</EditProfile>
`;
